exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroBatchReader.java

```diff
@@ -0,0 +1,368 @@
+        scalarWriter.setLong((long) value);
+        break;
+      case FLOAT4:
+        scalarWriter.setDouble((float) value);
+        break;
+      case FLOAT8:
+        scalarWriter.setDouble((double) value);
+        break;
+      case VARDECIMAL:
+        BigInteger bigInteger;
+        if (value instanceof ByteBuffer) {
+          ByteBuffer decBuf = (ByteBuffer) value;
+          bigInteger = new BigInteger(decBuf.array());
+        } else {
+          GenericFixed genericFixed = (GenericFixed) value;
+          bigInteger = new BigInteger(genericFixed.bytes());
+        }
+        BigDecimal decimalValue = new BigDecimal(bigInteger, columnMetadata.scale());
+        scalarWriter.setDecimal(decimalValue);
+        break;
+      case BIT:
+        scalarWriter.setBoolean((boolean) value);
+        break;
+      case VARCHAR:
+        byte[] binary;
+        int length;
+        if (value instanceof Utf8) {
+          Utf8 utf8 = (Utf8) value;
+          binary = utf8.getBytes();
+          length = utf8.getByteLength();
+        } else {
+          binary = value.toString().getBytes(Charsets.UTF_8);
+          length = binary.length;
+        }
+        scalarWriter.setBytes(binary, length);
+        break;
+      case VARBINARY:
+        if (value instanceof ByteBuffer) {
+          ByteBuffer buf = (ByteBuffer) value;
+          scalarWriter.setBytes(buf.array(), buf.remaining());
+        } else {
+          byte[] bytes = ((GenericFixed) value).bytes();
+          scalarWriter.setBytes(bytes, bytes.length);
+        }
+        break;
+      case TIMESTAMP:
+        String avroLogicalType = columnMetadata.property(AvroSchemaUtil.AVRO_LOGICAL_TYPE_PROPERTY);
+        if (AvroSchemaUtil.TIMESTAMP_MILLIS_LOGICAL_TYPE.equals(avroLogicalType)) {
+          scalarWriter.setLong((long) value);
+        } else {
+          scalarWriter.setLong((long) value / 1000);
+        }
+        break;
+      case DATE:
+        scalarWriter.setLong((int) value * (long) DateTimeConstants.MILLIS_PER_DAY);
+        break;
+      case TIME:
+        if (value instanceof Long) {
+          scalarWriter.setInt((int) ((long) value / 1000));
+        } else {
+          scalarWriter.setInt((int) value);
+        }
+        break;
+      case INTERVAL:
+        GenericFixed genericFixed = (GenericFixed) value;
+        IntBuffer intBuf = ByteBuffer
+          .wrap(genericFixed.bytes())
+          .order(ByteOrder.LITTLE_ENDIAN)
+          .asIntBuffer();
+
+        Period period = Period
+          .months(intBuf.get(0))
+          .withDays(intBuf.get(1))
+          .withMillis(intBuf.get(2));
+
+        scalarWriter.setPeriod(period);
+        break;
+      default:
+        throw UserException.dataReadError()
+          .message("Unexpected scalar schema type: ", columnMetadata.type())
+          .addContext("Column", columnMetadata)
+          .addContext("Reader", this)
+          .build(logger);
+    }
+  }
+
+  public static class AvroReaderConfig {
```

Comment:
The config is generally needed only if we pass more than one item to the reader. Here we just pass the plugin, so (IIRC) the class isn't necessary; though no harm in having it.